### ISSUE TITLE
Why do train_dataloader and val_dataloader load the same data during training?

### ISSUE THREAD

**Issue by zll-leo** [wokaikaixinxin/ai4rs#11]
I'm a newbie just getting into rotating frame detection. When running this project, I noticed that the training set and validation set are loaded with the same data in the configuration file. 
Doesn't this lead to data leakage? I'm curious why it's designed this way. I'd really appreciate any insights!

In the configuration file：projects/rotated_rtdetr/configs/dota_rr.py
``` python
data_root = 'data/split_ss_dota/' # 数据集根路径

train_dataloader = dict(
    batch_size=4,
    num_workers=4,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    batch_sampler=None,
    pin_memory=False,
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        ann_file='trainval/annfiles/',
        data_prefix=dict(img_path='trainval/images/'),
        filter_cfg=dict(filter_empty_gt=True),
        pipeline=train_pipeline))
val_dataloader = dict(
    batch_size=4,
    num_workers=4,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        ann_file='trainval/annfiles/',                                # 与train一样
        data_prefix=dict(img_path='trainval/images/'),  # 与train一样
        test_mode=True,
        pipeline=val_pipeline))

```

**Comment by wokaikaixinxin** [comment]
> I'm a newbie just getting into rotating frame detection. When running this project, I noticed that the training set and validation set are loaded with the same data in the configuration file. Doesn't this lead to data leakage? I'm curious why it's designed this way. I'd really appreciate any insights!
> 
> In the configuration file：projects/rotated_rtdetr/configs/dota_rr.py
> 
> data_root = 'data/split_ss_dota/' # 数据集根路径
> 
> train_dataloader = dict(
>     batch_size=4,
>     num_workers=4,
>     persistent_workers=True,
>     sampler=dict(type='DefaultSampler', shuffle=True),
>     batch_sampler=None,
>     pin_memory=False,
>     dataset=dict(
>         type=dataset_type,
>         data_root=data_root,
>         ann_file='trainval/annfiles/',
>         data_prefix=dict(img_path='trainval/images/'),
>         filter_cfg=dict(filter_empty_gt=True),
>         pipeline=train_pipeline))
> val_dataloader = dict(
>     batch_size=4,
>     num_workers=4,
>     persistent_workers=True,
>     drop_last=False,
>     sampler=dict(type='DefaultSampler', shuffle=False),
>     dataset=dict(
>         type=dataset_type,
>         data_root=data_root,
>         ann_file='trainval/annfiles/',                                # 与train一样
>         data_prefix=dict(img_path='trainval/images/'),  # 与train一样
>         test_mode=True,
>         pipeline=val_pipeline))

1）在rotated_rtdetr中，DOTA1.0数据集处理成了trainval 和 test集合。trainval集合有图像和对应标签。test集合只有图像，无标签。
2）train，validation，test三个环节不一样。
3）train 和 validation 环节都用的 trainval集合。
4）test环节(最终结果)是test集合上评测并上传官网得到的。所以不泄露数据。

**Comment by zll-leo** [comment]
感谢你的回复！
DOTA1.0数据集的处理和test环节我清楚了，但还有一个问题：在train和validation环节使用同样的集合，那如何判断模型是否收敛或过拟合？

**Comment by wokaikaixinxin** [comment]
> 感谢你的回复！ DOTA1.0数据集的处理和test环节我清楚了，但还有一个问题：在train和validation环节使用同样的集合，那如何判断模型是否收敛或过拟合？

1）凭经验。
2）虽然DOTA-v1.0在train和validation环节使用同样的集合，但精度是随着训练轮次增加逐步上升的。
3）在一定程度上确实存在不直观的问题。

**Comment by zll-leo** [comment]
明白了，谢谢
